paywall/src/hooks/browser/useListenForPostMessage.js

```diff
@@ -2,21 +2,32 @@ import { useEffect, useState } from 'react'
 import useConfig from '../utils/useConfig'
 import { getRouteFromWindow } from '../../utils/routes'
 
-export default function useListenForPostmessage(window) {
+export default function useListenForPostmessage(
+  window,
+  { type, validator = false, defaultValue }
+) {
   const { isInIframe, isServer } = useConfig()
   const parent = window && window.parent
-  const [data, setData] = useState()
+  const [data, setData] = useState(defaultValue)
   const saveData = event => {
     const { origin } = getRouteFromWindow(window)
     // ignore messages that do not come from our parent window
     if (event.source !== parent || event.origin !== origin) return
-    setData(event.data)
+    // data must be of shape { type: 'type', payload: <value> }
+    if (!event.data || !event.data.type) return
+    if (!event.data.hasOwnProperty('payload')) return
+    if (typeof event.data.type !== 'string') return
+    if (event.data.type !== type) return
+    // optional validator
+    if (!validator || (validator && validator(event.data.payload))) {
```

Comment:
`validator` is a function that returns truthy for data we want? Do you have an example of how that would be used in practice? Just curious, it seems like a good thing to have but I can't come up with anything concrete at the moment.

LG!

Reply:
yep. The most immediate use is ensuring that scrollPosition is always a positive integer, so the existing `unlock-app/utils/validators.js` function `isPositiveInteger` can be used as a validator in that instance. That's why I chose this specific example in the description:

```js
const scrollPosition = useListenForPostMessage(window, {
  type: POST_MESSAGE_SCROLL_POSITION,
  validator: isPositiveInteger, // this function only validates numbers >= 0
  default: 0
})
```

Reply:
D'oh, can't believe I skimmed right over that. Makes sense!